pkg/cli/args.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	libflag "flag"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	SingleDashFlag NormalizeActsType = iota
+	DoubleDashFlag NormalizeActsType = iota
+)
+
+var (
+	singleDashRegexp = regexp.MustCompile(`^-[^-]`)
+	doubleDashRegexp = regexp.MustCompile(`^--[^-]`)
+)
+
+type NormalizeActsType byte
+
+// Args is a wrapper for `urfave`'s `cli.Args` interface, that provides convenient access to CLI arguments.
+type Args struct {
+	cli.Args
+}
+
+func newArgs(args []string) *Args {
+	// This is the only way to avoid duplicating code from the private struct `urfave`'s `cli.args` that implements `cli.Args` interface.
+	flagSet := libflag.NewFlagSet("", libflag.ContinueOnError)
+	flagSet.SetOutput(io.Discard)
+	flagSet.Parse(append([]string{"--"}, args...))
```

Comment:
Error returned from `Parse()` seems to be unhandled...
```
func (f *FlagSet) Parse(arguments []string) error
```

Reply:
Redone to get rid of this feature altogether